pyproject.toml

```diff
@@ -36,37 +36,42 @@ exclude = ['*tests', '*testdata']
 
 [tool.poetry.dependencies]
 python = ">=3.8,<3.11"
-beartype = "^0.10.4"
 numpy = "^1.23.0"
 scipy = "^1.9.0"
-networkx = "^2.8.3"
 scikit-learn = "^1.1.0"
 pandas = "^1.1"
 importlib-resources = { version = "*", python = "<3.9" }
-graphs = { git = "https://github.com/py-why/graphs.git", branch = 'main', optional = true }
+networkx = { git = "https://github.com/adam2392/networkx.git", branch = 'mixededge' }
 pywhy-graphs = { git = "https://github.com/py-why/pywhy-graphs.git", branch = 'main', optional = true }
-# pywhy-graphs = { path = "../pywhy-graphs", develop = true }
 pygraphviz = { version = "*", optional = true }
 
-# TODO: the below needs to be refactored into separate dev-dependencies
-# https://github.com/python-poetry/poetry/is
-# it is inefficient to have to install every single dependencey to run
-# e.g. a style check.
-[tool.poetry.dev-dependencies]
-pytest = "^7.1.2"
-pytest-cov = "^3.0.0"
-mypy = "^0.971"
+[tool.poetry.group.style]
+optional = true
+[tool.poetry.group.style.dependencies]
 poethepoet = "^0.16.0"
+mypy = "^0.971"
 black = "^22.6.0"
 isort = "^5.10.1"
 flake8 = "^5.0.4"
```

Comment:
flake8, bandit, and pylint should probably be in a new "linting" group

Reply:
Since they're part of the same style CI job, is this necessary? The `style` group is for devs that want to PR code but say not build the docs, or run tests. Open to changing tho. Let me merge this in first, so I can test out the downstream PRs.